src/Components/Activities/Form/index.jsx

```diff
@@ -2,100 +2,49 @@ import React, { useEffect, useState } from 'react';
 import styles from './form.module.css';
 import { Input, TextArea } from '../../Shared/Inputs';
 import { Link, useParams, useLocation, useHistory } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { postActivity, putActivity } from '../../../redux/activities/thunks';
 import Button from '../../Shared/Button';
 import Modal from '../../Shared/Modal';
 
 const Form = () => {
+  const { data, message, success, error } = useSelector((state) => state.activities);
   const { id } = useParams();
   const [name, setName] = useState('');
   const [description, setDescription] = useState('');
   const [isActive, setIsActive] = useState(false);
-  const [message, setMessage] = useState('');
   const [showModal, setShowModal] = useState(false);
   const location = useLocation();
   const history = useHistory();
+  const dispatch = useDispatch();
 
-  const onRedirect = {
-    pathname: '/activities',
-    state: { message: '' }
+  const handleModal = () => {
+    setShowModal(!showModal);
+    setTimeout(() => {
+      setShowModal();
+    }, 4000);
```

Comment:
this timeout should be set to 2000 like the other ones

Reply:
This timeout is for the error messages, it last a little longer so the user can read the error.